src/NHSD.BuyingCatalogue.Organisations.Api/Repositories/OdsRepository.cs

```diff
@@ -14,19 +15,23 @@ namespace NHSD.BuyingCatalogue.Organisations.Api.Repositories
     internal sealed class OdsRepository : IOdsRepository
     {
         private readonly OdsSettings settings;
+        private readonly IAppCache appCache;
 
-        public OdsRepository(OdsSettings settings)
+        public OdsRepository(OdsSettings settings, IAppCache appCache)
         {
             this.settings = settings ?? throw new ArgumentNullException(nameof(settings));
+            this.appCache = appCache;
         }
 
         public async Task<OdsOrganisation> GetBuyerOrganisationByOdsCodeAsync(string odsCode)
         {
-            var response = await settings.ApiBaseUrl
-                .AppendPathSegment("organisations")
-                .AppendPathSegment(odsCode)
-                .AllowHttpStatus("3xx,4xx")
-                .GetJsonAsync<OdsResponse>();
+            var response = await appCache.GetOrAddAsync(
+                $"Ods-Org-{odsCode}",
```

Comment:
Is there a reason why we can't just use the ODS code as the key given that we're not currently using any other type of key?

Reply:
changed